api/v1alpha8/openstackcluster_types.go

```diff
@@ -167,6 +167,8 @@ type OpenStackClusterSpec struct {
 
 // OpenStackClusterStatus defines the observed state of OpenStackCluster.
 type OpenStackClusterStatus struct {
+	// Ready is true when the cluster infrastructure is ready.
+	// +optional
```

Comment:
This sounds odd to me that we don't `omitempty` but I've kept it consistent with `OpenStackMachine.Status.Ready`. I'm happy to do it another way, like add `omitempty` to the `OpenStackMachine.Status.Ready` too or ... ?

Reply:
maybe you can propose another PR for that purpose later if we decide `omitempty` is needed :)

Reply:
ok, let me split it. I did it within the same PR because I saw the second error right after the warning, so I thought I would fix them together but I think it's best to split.

Reply:
done via https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1842